providers/implementations/keymgmt/ml_dsa_kmgmt.c

```diff
@@ -251,9 +251,15 @@ static int ml_dsa_key_fromdata(ML_DSA_KEY *key, const OSSL_PARAM params[],
     }
 
     /* Error if the supplied public key does not match the generated key */
-    return pk_len == 0
+    if (pk_len == 0
         || seed_len + sk_len == 0
-        || memcmp(ossl_ml_dsa_key_get_pub(key), pk, pk_len) == 0;
+        || memcmp(ossl_ml_dsa_key_get_pub(key), pk, pk_len) == 0)
+        return 1;
+    ERR_raise_data(ERR_LIB_PROV, PROV_R_INVALID_KEY,
+                   "explicit %s public key does not match private",
+                   key_params->alg);
+    ossl_ml_dsa_key_reset(key);
```

Comment:
This reset is also redundant.

Reply:
Well, import takes a key argument, with the key created and freed *separately*, so we should not assume that the key is always freed when import fails. The explicit `reset()` is not expensive, so I think it should be done as future-proofing, even if key reuse after a failed import is difficult at present.